MatomoTracker/MatomoTracker.swift

```diff
@@ -468,3 +468,24 @@ extension MatomoTracker {
     /// The version of the Matomo SDKs
     @objc public static let sdkVersion = "7.3"
 }
+
+extension MatomoTracker {
+    /// Resets all session, visitor and campaign information.
+    ///
+    /// Dispatches events before restting itself.
+    /// After calling this method this instance behaves like the app has been freshly installed.
+    public func reset() {
+        dispatch()
+        
+        matomoUserDefaults.reset()
+        
+        visitor = Visitor.current(in: matomoUserDefaults)
+        session = Session.current(in: matomoUserDefaults)
+        dimensions = []
+        customVariables = []
+        campaignName = nil
+        campaignKeyword = nil
+        
+        startNewSession()
+    }
+}
```

Comment:
With this function we do have the risk, that if we add any properties, we forget to reset them in here as well. Unfortunately I don't have a great idea how to prevent this. I still wanted to bring this up here in case you have an idea.

One option would certainly be to replace the instance of `MatomoTracker`. But since the SDK doesn't provide any instance, this would be left for the user and all we could offer in this function here would be to reset the persisted data. I don't think that's great.